stable/pomerium/templates/tls-secrets.yaml

```diff
@@ -16,3 +16,138 @@ data:
 ---
 {{- end }}
 {{- end }}
+
+{{- define "pomerium.authenticate.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.forceGenerateTLS }}
+    helm.sh/hook: pre-upgrade
+{{- else if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.authenticate.tlsSecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/tls
+data:
+{{- end -}}
+
+{{- define "pomerium.authorize.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.forceGenerateTLS }}
+    helm.sh/hook: pre-upgrade
+{{- else if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.authorize.tlsSecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/tls
+data:
+{{- end -}}
+
+{{- define "pomerium.proxy.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.forceGenerateTLS }}
+    helm.sh/hook: pre-upgrade
+{{- else if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.proxy.tlsSecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: kubernetes.io/tls
+data:
+{{- end -}}
+
+{{- define "pomerium.ca.tlsSecretObject" -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ template "pomerium.name" . }}
+    helm.sh/chart: {{ template "pomerium.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+  annotations:
+{{- if .Values.config.forceGenerateTLS }}
+    helm.sh/hook: pre-upgrade
+{{- else if .Values.config.generateTLS }}
+    helm.sh/hook: pre-install
+{{- end }}
+  name: {{ template "pomerium.caSecret.name" . }}
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+data:
+{{- end -}}
+
+{{- /* TODO in future: Remove legacy logic */ -}}
+{{- if not .Values.existingLegacyTLSSecret }}
+{{- if or (and .Values.config.generateTLS .Release.IsInstall) (and .Values.config.generateTLS .Values.config.forceGenerateTLS) }}
```

Comment:
@travisgroth there seems to be an inconsistency here: in the values.yaml and README, `forceGenerateTLS` is not under `config`.

Reply:
Should be fixed

Reply:
@travisgroth some more optional proposed changes:
- have `existingLegacyTLSSecret` in `config` too to be consistent
- rephrase the condition above to: `if and .Values.config.generateTLS (or .Release.IsInstall .Values.config.forceGenerateTLS)`, the intent seems clearer with that
- indent this same condition once to make visible the various branches (and thus indent once more the conditions that are already indented below

With all of this, I think this will make this template much easier to read, understand and maintain :)
- move the first `$kp :=` below next to its first usage before `template pomerium.authenticate.tlsSecretObject`

Reply:
All good ideas.  Done.  👍